### PR TITLE
fix(form-service): add ability to summarize complex objects

### PR DESCRIPTION
**Pull request by rstyan**

### CODE REVIEW

File: libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts [hunk 1]
```diff
@@ -1,35 +1,107 @@
...
+const isNestedValue = (value: NestedStringArray): boolean => {
+  return Array.isArray(value) && value.length === 2 && typeof value[0] === 'string' && Array.isArray(value[1]);
+};
+
+/*
+ * Convert ['name', [['first', 'fred'], ['middle', 'jolly'], ['last', 'mcguire']]]
+ * into [['first', 'fred'], ['middle', 'jolly'], ['last', 'mcguire']]
+ */
+const flatten = (arr: NestedStringArray): string[][] => {
+  console.log('flattening');
```
Comment (athena-chen-chen) [inline libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts:67]: Please remove debugging code.

File: libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts [hunk 2]
```diff
@@ -1,35 +1,107 @@
-export const resolveLabelFromScope = (scope: string) => {
+import { LabelDescription } from '@jsonforms/core';
+
+type jsonformsLabel = string | boolean | LabelDescription | undefined;
+export const labelToString = (label: jsonformsLabel, scope: string): string => {
+  if (!label) {
+    return resolveLabelFromScope(scope);
+  }
+  if (typeof label === 'object' && label !== null && 'show' in label && label.show) {
```
Comment (athena-chen-chen) [inline libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts:8]: if (typeof label === 'object' && label?.show)  is good enough?

File: libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts (hunk 1 above)
Comment (thdlam) [inline libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts:67]: remove these console log/ console.dir

Comment (solittlework) [reply libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts:67]: Hi @rstyan, just a kind reminder that the jsonform/core includes the lodash lib, which means you can use the lodash flatten, isArray, get or other operations without enlarging the lib size.

File: libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts (hunk 2 above)
Comment (rstyan) [reply libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts:8]: No.  On a general object, you first have to test if 'show' is actually in the object, other wise you will get a compile-time error e.g _label?.show_ will have an error something like: "show is not a property of label".  label?.show is a run-time test,  checking to see if label is null before accessing its (known) field.

**Review by solittlework — APPROVED** [review]

**Pull request merged**